packages/react-charting/src/utilities/utilities.ts

```diff
@@ -1095,22 +1099,127 @@ export function wrapTextInsideDonut(selectorClass: string, maxWidth: number) {
       .attr('y', y)
       .attr('dy', lineNumber++ * lineHeight + 'em');
 
+    // tspanEllipsis is used for checking if truncation is required vertically/horizontally
+    // before appending the text to the inner donut text
+    let tspanEllipsis = d3Select('#Donut_center_text').text(null).append('tspan').attr('opacity', 0);
+
+    // Determine the ellipsis length for word truncation.
+    tspanEllipsis.text(ellipsis);
+    const ellipsisLength = tspanEllipsis.node()!.getComputedTextLength();
+    tspanEllipsis.text(null);
+
+    // Value concatinates and saves the final truncated string and returns it back to the donut chart
+    // to handle mouse over and mouse out scenarios
+    value = '';
+
     while ((word = words.pop()!)) {
       line.push(word);
       tspan.text(line.join(' ') + ' ');
-      if (tspan.node()!.getComputedTextLength() > maxWidth && line.length > 1) {
-        line.pop();
-        tspan.text(line.join(' ') + ' ');
-        line = [word];
-        tspan = text
-          .append('tspan')
-          .attr('id', `WordBreakId-${idx}-${lineNumber}`)
-          .attr('x', 0)
-          .attr('y', y)
-          .attr('dy', lineNumber++ * lineHeight + 'em')
-          .text(word);
+      tspanEllipsis.text(line.join(' ') + ' ');
+
+      // Determine if wrapping is required
+      if (tspan.node()!.getComputedTextLength() > maxWidth - ellipsisLength && line.length > 1) {
+        while (tspan.node()!.getComputedTextLength() > maxWidth - ellipsisLength) {
+          line.pop();
+          tspan.text(line.join(' ') + ' ');
+          tspanEllipsis.text(line.join(' ') + ' ');
+        }
+        // Determine if truncation is required vertically
+        // If truncation is not required vertically, append a new line while taking care of horizontal truncation
+        if (tspan.node()!.getBoundingClientRect().y < maxWidth) {
+          line = [word];
+          tspanEllipsis.text(word);
+
+          // Determine if truncation is appending the text exceeds maximum width vertically or horizontally
```

Comment:
what do you mean by verticl width

Reply:
I meant the vertical height, changing the sentence